.github/workflows/tests.yml

```diff
@@ -61,9 +64,48 @@ jobs:
               return true
             }
             console.log('CI-syntax-only label found. Skipping tests.')
+            return false
+
+      - name: Check for CI-no-fail-fast label
+        id: check-fail-fast
+        uses: actions/github-script@v3
+        if: github.event_name == 'pull_request'
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { data: { labels: labels } } = await github.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            })
+            if (!labels.map(label => label.name).includes('CI-no-fail-fast')) {
+              console.log('No CI-no-fail-fast label found.')
+              return true
+            }
+            console.log('CI-no-fail-fast label found.')
+            return false
+
+      - name: Check for CI-long-timeout label
+        id: check-timeout-minutes
+        uses: actions/github-script@v3
+        if: github.event_name == 'pull_request'
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { data: { labels: labels } } = await github.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            })
+            if (!labels.map(label => label.name).includes('CI-long-timeout)) {
+              console.log('No CI-long-timeout label found.')
+              return 60
```

Comment:
I'm of two minds about a 60-minute timeout. On the one hand, I think that there are enough formulae that actually take longer than an hour that it would be annoying to remember to label them (if you even remember they need labelling until it's too late).

On the other, I think this timeout is exactly what we need to prevent huge CI queues.

Reply:
Though it looks like there's a syntax error around here.

Reply:
I'm open to this being higher to start with